src/hooks/gatewayAuthToken/useGatewayAuthToken.ts

```diff
@@ -71,33 +102,22 @@ const useGatewayAuthToken = (prefixes: string[] | null) => {
         }
     }
 
-    // Grab the Gateway token from local storage
-    let jwt: JWTPayload | undefined;
-    try {
-        const gatewayConfig = getStoredGatewayAuthConfig();
-        jwt = gatewayConfig ? decodeJwt(gatewayConfig.token) : undefined;
-    } catch {
-        jwt = undefined;
-    }
-
-    // Check if the Gateway token is expired and we need to get a new one
-    let tokenExpired = true;
-    if (jwt?.exp) {
-        tokenExpired = isBefore(jwt.exp * 1000, Date.now());
-    }
-
     const { data, mutate } = useSWR(
-        tokenExpired || authorized_prefixes.length > 0
+        hasAuthorizedPrefixes && cacheKey
```

Comment:
We can now just call as long as we have prefixes to fetch and a key to save it as. Mainly checking both to just be super safe even though if we have prefixes then we should have a cache key